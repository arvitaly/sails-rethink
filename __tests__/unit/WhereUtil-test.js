"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const WhereUtil_1 = require("./../../WhereUtil");
describe("WhereUtil tests", () => {
    it("when criteria is plain object, should expect", () => {
        const expr = WhereUtil_1.findCriteriaToExpr({
            field1: { lessThan: 15 },
            field2: { lessThanOrEqual: 25 },
            field3: { greaterThan: 40.8 },
            field4: { greaterThanOrEqual: "xo", like: "m16575" },
            field5: { not: new Date(), like: "%hel%l%o", contains: "cont1", startsWith: "ho", endsWith: "ar" },
        });
        expect(expr.toString()).toMatchSnapshot();
    });
    it("when criteria is or, should expect", () => {
        const expr = WhereUtil_1.findCriteriaToExpr({
            or: [{
                    test: "x",
                    field: 1.1,
                }, {
                    m: 16,
                    m2: "f1"
                }],
        });
        expect(expr.toString()).toMatchSnapshot();
    });
});
