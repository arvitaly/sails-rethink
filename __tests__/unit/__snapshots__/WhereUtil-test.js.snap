// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhereUtil tests when criteria is or, should expect 1`] = `"r.row(\\"test\\").downcase().eq(\\"x\\").and(r.row(\\"field\\")).or(r.row(\\"m\\").and(r.row(\\"m2\\").downcase().eq(\\"f1\\")))"`;

exports[`WhereUtil tests when criteria is plain object, should expect 1`] = `"r.row(\\"field1\\").lt(15).and(r.row(\\"field2\\").le(25)).and(r.row(\\"field3\\").gt(40.8)).and(r.row(\\"field4\\").downcase().ge(\\"xo\\").and(r.row(\\"field4\\").downcase().downcase().match(\\"^m16575$\\"))).and(r.row(\\"field5\\").not().and(r.row(\\"field5\\").downcase().match(\\"^.*hel.*l.*o$\\")).and(r.row(\\"field5\\").downcase().downcase().match(\\"cont1\\")).and(r.row(\\"field5\\").downcase().downcase().downcase().match(\\"^ho\\")).and(r.row(\\"field5\\").downcase().downcase().downcase().downcase().match(\\"ar$\\")))"`;
